src/bootstrap/lib.rs

```diff
@@ -811,6 +811,11 @@ impl Build {
         if target.contains("apple-darwin") {
             base.push("-stdlib=libc++".into());
         }
+        
+        // Required for LLVM to properly compile.
+        if target.contains("apple-ios") {
+            base.push("-miphoneos-version-min=10.0".into());
```

Comment:
This seems like a pretty recent iOS version -- depending on why LLVM isn't compiling (not sure what "properly" means), maybe we should be setting this to something older?

Also, it feels like this should go into src/bootstrap/native.rs llvm step if it's LLVM-specific.

Reply:
Nope, `10.0` is the minimum required for some C++ features.

Example error with lower target: 
![image](https://user-images.githubusercontent.com/65794972/86179712-cb149180-baf8-11ea-85f3-f72199eb0a7f.png)

Reply:
Okay well -- I guess maybe we do need 10.0. But even with that, feels like this should go in LLVM-specific cflags rather than being set globally, if possible, to at least avoid e.g. std's C dependencies (libbacktrace as of now I think) getting polluted with this.

Also, would be nice to avoid images and copy/paste text (perhaps into a gist), makes it much easier to read.

Reply:
this is an old image, i don't feel like waiting 30 minutes in a macOS VM just for this specific error.

Also, cc-rs is what sets `-miphoneos-version-min` to 7.0. There is no way to override this without setting an environmental variable (not to cmake, to whatever is running cc-rs) or passing it yourself.

Reply:
Resolved in https://github.com/rust-lang/rust/pull/73903/commits/c22bcb03814a1600d17097123813495ea0a40796